README.md

```diff
@@ -88,29 +137,113 @@ const PersonSchema = z.object({
+
+### Create Your Own Transformer
+
+Instead of using the opinionated `Fixture` class, you can extend the unopinionated `Transformer` and register the desired generators.
+
+```ts
+import { Transformer } from 'zod-fixture';
+
+const transform = new Transformer().extend([
+	/* insert your generators here */
+]);
+
+const value = transform.from(zodSchema);
+```
+
+## API
+
+### Generators
+
+To generate a value based on a zod type we're using what we call a `Generator`.
+
+To make your own generators simpler this library also includes some useful utility methods to generate data.
+
+In the example below we create our own `NumberBetween0And25Generator` to return more realastic numbers using the `random` utilities.
+
+```ts
+import { z, ZodObject } from 'zod';
+import { createFixture, Generator } from 'zod-fixture';
+
+const NumberBetween0And25Generator = Generator({
+	// we're interested in zod numbers
+	schema: ZodNumber,
+	// because we only have one number object we can ignore the additional `filter`
+	// we make use of the utils to create a random number
+	output: ({ core }) => core.utils.random.int({ min: 0, max: 25 }),
+});
+
+const AddressGenerator = Generator({
+	// we're interested in zod objects
+	schema: ZodObject,
+	// we only want to change the behavior of the address object
+	filter: ({ context }) => context.path.at(0) === 'address',
+	// return the desired output based on a custom implementation
+	output: () => ({
+		street: 'My Street',
+		city: 'My City',
+		state: 'My State',
+	}),
+});
+
+const PersonSchema = z.object({
+	name: z.string(),
+	birthday: z.date(),
+	address: z.object({
+		street: z.string(),
+		city: z.string(),
+		state: z.string(),
+	}),
+	pets: z.array(z.object({ name: z.string(), breed: z.string() })),
+	totalVisits: z.number(),
+});
+
+const person = new Fixture()
+	.extend([AddressGenerator, NumberBetween0And25Generator])
+	.from(PersonSchema);
+```
+
+When we create a new `person` fixture using the two custom generators we get the following value (notice the `address` and `totalVisits` values).
 
 ```json
 {
-	"name": "name-e5d36b60-2404-4cdc-a438-e9fc8bf9b704",
-	"birthday": "2022-04-01T07:22:37.187Z",
-	"address": { "street": "My Street", "city": "My City", "state": "My State" },
+	"address": {
+		"city": "My City",
+		"state": "My State",
+		"street": "My Street"
+	},
+	"birthday": "1926-02-23T02:07:24.494Z",
+	"name": "c",
 	"pets": [
 		{
-			"name": "name-1cd9b68f-94ab-4881-b5ba-2a996b68d9c3",
-			"breed": "breed-020aae4b-c88f-4f7e-9711-bbe1b5a01c0f"
+			"breed": "5yOQfkYfI6=kRuH^F?5BCNHft",
+			"name": "mYxRp1GBY2aw"
+		},
+		{
+			"breed": "6Qz\\\\s",
+			"name": "_"
+		},
+		{
+			"breed": "6e9",
+			"name": ";l]@"
 		}
 	],
-	"totalVisits": 3
+	"totalVisits": 22
 }
 ```
 
-## Getting started with GitHub Codespaces
+⚠️ Note: The order the registered generators matters. The first generator that matches the conditions (`schema` and `filter`) is used to create the value.
```

Comment:
Because we're using vitepress we can make this look nice.

```suggestion
::: tip
The order the registered generators matters. The first generator that matches the conditions (`schema` and `filter`) is used to create the value.
:::
```